### ISSUE TITLE
is_normal does not meet the requirements

### ISSUE THREAD

**Issue by benruijl** [jkarns275/f128#25]
According to the docs for f64, `is_normal` should meet the following contract:
```
Returns true if the number is neither zero, infinite, subnormal, or NaN.

use num_traits::Float;
use std::f32;

let min = f32::MIN_POSITIVE; // 1.17549435e-38f32
let max = f32::MAX;
let lower_than_min = 1.0e-40_f32;
let zero = 0.0f32;

assert!(min.is_normal());
assert!(max.is_normal());

assert!(!zero.is_normal());
assert!(!f32::NAN.is_normal());
assert!(!f32::INFINITY.is_normal());
// Values between `0` and `min` are Subnormal.
assert!(!lower_than_min.is_normal());
```

A partial port to f128 reveals that it does not meet the requirements for at least `MIN_POSITIVE` and `MAX`:
```
    fn test_is_normal() {
        let min = f128::MIN_POSITIVE;
        let max = f128::MAX;
        let zero = 0.0f32;
        let one = f128::ONE;
        let minone = -f128::ONE;

        assert!(one.is_normal());
        assert!(minone.is_normal());
        assert!(min.is_normal()); // fails
        assert!(max.is_normal()); //fails

        assert!(!zero.is_normal());
        assert!(!f128::NAN.is_normal());
        assert!(!f128::INFINITY.is_normal());
    }
```

**Comment by jkarns275** [comment]
I've fixed the implementation of `is_normal` 3e8670cddfd642e2a18551b49c557264db3da33b , should be correct now. Can you verify?

**Comment by jkarns275** [comment]
Okay this was caused by a misunderstanding on my part. Float::MIN_POSITIVE is supposed to be the smallest normal number not the smallest positive representable number. I fixed this and added two additional constants for clarity. The latest version of the crate passes the test you provided.

**Comment by jkarns275** [comment]
fixed here 53da120443372cafc3bb0dd2ec41d813343ace43
